MetasysServices/Models/MetasysObject.cs

```diff
@@ -84,12 +84,30 @@ public class MetasysObject : Utils.ObjectUtil
         public string Classification { get; set; }
         private const string DefaultClassification = "object";
 
+        /// <summary>
+        /// Create a tree of objects out of a root node
+        /// </summary>
+        /// <param name="token"></param>
+        /// <param name="version"></param>
+        internal MetasysObject(JToken token, ApiVersion version)
```

Comment:
This new constructor can take a node (with children) and construct the appropriate tree out of it. I think the existing constructor can maybe be removed after some refactoring. But I wanted to minimize changes.